### PR TITLE
Introduce standarised Data Sampling condition

### PR DESCRIPTION
**Pull request by knopers8**
There were feature requests concerning the way of taking decisions about sampling data - some users would like to receive not only random samples, but f.e. 3 consecutive during some periods. This commit prepares place for such a need.
DataSamplingCondition is not used yet by Dispatchers, but it will be when DataSamplingPolicy is implemented.
@Barthelemy

### CODE REVIEW

**Comment by Barthelemy** [comment]
I will have a look asap

**Comment by knopers8** [comment]
Note that also filters can be implemented as `DataSamplingCondition`s and, if needed, we may provide possibility to combine multiple conditions using `any_of` or `all_of`.

File: Framework/Core/src/DataSamplingConditionRandom.cxx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+/// \file DataSamplingConditionRandom.cxx
+/// \brief Implementation of random DataSamplingCondition
+///
+/// \author Piotr Konopka, piotr.jan.konopka@cern.ch
+
+#include "Framework/DataSamplingCondition.h"
+#include "Framework/DataProcessingHeader.h"
+
+#include <TRandom3.h>
```
Comment (Barthelemy) [inline Framework/Core/src/DataSamplingConditionRandom.cxx]: I usually use TRandom3 as well, but just for your information there are many other such classes in ROOT : https://root.cern.ch/doc/v612/group__Random.html

File: Framework/Core/src/DataSamplingConditionRandom.cxx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace o2
+{
+namespace framework
+{
+
+// todo: choose the best PRNG (TRandom3 is fast, but i am not sure about its statistical soundness and behaviour with
```
Comment (Barthelemy) [inline Framework/Core/src/DataSamplingConditionRandom.cxx]: I see that my comment above is already addressed in this todo

File: Framework/Core/src/DataSamplingConditionRandom.cxx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+// very small percents).
+// todo: consider using run number as a seed
+
+using namespace o2::header;
+
+/// \brief A DataSamplingCondition which makes decicions randomly, but with determinism.
```
Comment (Barthelemy) [inline Framework/Core/src/DataSamplingConditionRandom.cxx]: decicions -> decisions

File: Framework/Core/include/Framework/DataSamplingCondition.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  virtual ~DataSamplingCondition() = default;
+
+  /// \brief Reads custom configuration parameters.
+  virtual void configure(const o2::configuration::tree::Branch&) = 0;
+  /// \brief Makes decision whether to pass a data sample or not.
+  virtual bool decide(const o2::framework::DataRef&) = 0;
+
+  // a list of getters of specific DataSamplingCondition's
+  /// \brief Getter for DataSamplingConditionRandom
+  static std::unique_ptr<DataSamplingCondition> getDataSamplingConditionRandom();
```
Comment (Barthelemy) [inline Framework/Core/include/Framework/DataSamplingCondition.h]: Good for now, if there were many more one could consider a separate Factory class (here the parent class acts as a Factory, which I like).

**Comment by Barthelemy** [comment]
@sawenzel the failing tests have nothing to do with this code. Could you force squash and merge ?

**Comment by Barthelemy** [comment]
@dberzano  the failing tests have nothing to do with this code. Could you force squash and merge ?

**Pull request merged**